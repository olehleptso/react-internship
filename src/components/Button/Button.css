.primary {
    font-family: 'Montserrat', sans-serif;
    padding: 0;
    cursor: pointer;
    width: 40%;
    height: 44px;
    background-color: #10c9c3;
    border: 0;
    font-size: 12px;
    font-weight: 500;
    text-transform: uppercase;
    letter-spacing: 1.2px;
    color: white;
    outline: none; 
}

.primary:hover {
    background: white;
    color: #10c9c3;
    box-shadow: inset 0 0 0 1px #10c9c3;
}