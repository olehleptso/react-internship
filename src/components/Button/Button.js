import './Button.css';
import PropTypes from 'prop-types';

function Button ({text, styles, onclick}) {
    return (
            <button className={styles} onClick={onclick}>{text}</button>
    )
}

export default Button;

Button.propTypes = {
    text: PropTypes.string,
    styles: PropTypes.string
};