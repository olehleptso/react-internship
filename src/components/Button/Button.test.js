import { render, screen } from '@testing-library/react';
import Button from './Button';

test('renders search button component', () => {
    const { asFragment } = render(<Button styles="primary" text="Hello" />);
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });