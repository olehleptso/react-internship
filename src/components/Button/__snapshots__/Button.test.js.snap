// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search button component 1`] = `
<DocumentFragment>
  <button
    class="primary"
  >
    Hello
  </button>
</DocumentFragment>
`;
